vertx-mongo-client/src/main/java/io/vertx/ext/mongo/impl/config/WriteConcernParser.java

```diff
@@ -11,51 +12,60 @@
 class WriteConcernParser {
   private final WriteConcern writeConcern;
 
-  public WriteConcernParser(JsonObject config) {
-    // Allow convenient string value for writeConcern e.g. ACKNOWLEDGED, SAFE, MAJORITY, etc
-    WriteConcern wc;
-    String wcs = config.getString("writeConcern");
-    if (wcs != null) {
-      wc = WriteConcern.valueOf(wcs);
-      if (wc == null) throw new IllegalArgumentException("Invalid WriteConcern " + wcs);
+  WriteConcernParser(ConnectionString connectionString, JsonObject config) {
```

Comment:
Github actually lets you do that by appending ?w=1

Cheers,
Martijn

On 24 May 2016 at 09:47, meggarr <notifications@github.com> wrote:

> In
> vertx-mongo-client/src/main/java/io/vertx/ext/mongo/impl/config/WriteConcernParser.java
> <https://github.com/vert-x3/vertx-mongo-client/pull/63#discussion_r64350636>
> :
>
> > @@ -11,51 +12,60 @@
> >  class WriteConcernParser {
> >    private final WriteConcern writeConcern;
> >
> > -  public WriteConcernParser(JsonObject config) {
> > -    // Allow convenient string value for writeConcern e.g. ACKNOWLEDGED, SAFE, MAJORITY, etc
> > -    WriteConcern wc;
> > -    String wcs = config.getString("writeConcern");
> > -    if (wcs != null) {
> > -      wc = WriteConcern.valueOf(wcs);
> > -      if (wc == null) throw new IllegalArgumentException("Invalid WriteConcern " + wcs);
> > +  WriteConcernParser(ConnectionString connectionString, JsonObject config) {
>
> I would suggest to pull down the code change and using *gitk* to view and ignore
> space changes, that will make it clearer.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/vert-x3/vertx-mongo-client/pull/63/files/c54f1509e6f9a29bc687129e3377f1fc07825f7d#r64350636>
>